torchvision/transforms/v2/_augment.py

```diff
@@ -135,3 +138,150 @@ def _transform(
             inpt = F.erase(inpt, **params, inplace=self.inplace)
 
         return inpt
+
+
+class _BaseMixupCutmix(Transform):
+    def __init__(self, *, alpha: float, num_categories: Optional[int] = None, labels_getter="default") -> None:
+        super().__init__()
+        self.alpha = alpha
+        self._dist = torch.distributions.Beta(torch.tensor([alpha]), torch.tensor([alpha]))
+
+        self.num_categories = num_categories
+
+        self.labels_getter = labels_getter
+        self._labels_getter = _parse_labels_getter(labels_getter)
+
+    def forward(self, *inputs):
+        inputs = inputs if len(inputs) > 1 else inputs[0]
+        flat_inputs, spec = tree_flatten(inputs)
+        needs_transform_list = self._needs_transform_list(flat_inputs)
```

Comment:
Do we really need this `needs_transform_list` ?
We can know where labels are and we pass through non-handled types, it feels like we can just transforms all items of the input ?

Reply:
See https://github.com/pytorch/vision/pull/7731#discussion_r1259519124

Reply:
Linked comment does not respond to my comment. As we talked previously, the way how it is coded here, we do not need to create redundant `needs_transform_list` structure and can pass all flat inputs items into `_get_params` and `_transform`. Those functions are handling unsupported types internally.

Reply:
As discussed offline, this is needed. Imagine a sample like

```python
sample = {
    "image": datapoints.Image(...),
    "label": torch.Tensor(...),
    "some_other_thing": torch.Tensor(...),
}
```

Without `needs_transform_list`, we cannot prevent `"some_other_thing"` to be transformed as an image.